Small improvements

tlipski opened the pull request:
new AngularJS selectors
webdriver/jvm timeout fixtures
make fill-form smarter
add url prefix support

tlipski committed:
add single quote in attribute value support to by-attribute selector

tlipski committed:
add by-ng-hide and by-ng-show selectors

tlipski committed:
add webdriver/jvm timeout fixtures

tlipski committed:
make fill-form support non-text controls (checkbox, radio, select) and wait for the control to be enabled before setting the value and take input order into account if provided

tlipski committed:
add url prefix resolution support from system properties/environment variables - useful for example for testing using bamboo or against AWS instances

kamituel commented on src/io/aviso/taxi_toolkit/fixtures.clj:
```diff
@@ -0,0 +1,74 @@
+                       (Thread/sleep (* timeout-seconds 1000))
+                       (when-not @finished?
+                         (println "Timeout after " timeout-seconds "seconds.")
+                         (System/exit -100))))
```
Negative return value? Not all shell's support that I think.

kamituel replied:
Yep, consider this:

``` bash
$ cat a.sh
exit -100
$ sh a.sh
$ echo $?
156
```

tlipski replied:
Right, I will use 99 then

kamituel commented on src/io/aviso/taxi_toolkit/fixtures.clj:
```diff
@@ -0,0 +1,74 @@
+            [clj-webdriver.remote.server :as webdriver-remote])
+  (:import (java.util.concurrent TimeUnit)))
+
+(defn webdriver-remote-fixture
```
Could this fn be somehow separated in two pieces, one general for setting up a driver, timeouts etc., and the other one that would build on top of that with remote session stuff?

tlipski replied:
Yeah, I will split it even into 3 functions

kamituel commented on src/io/aviso/taxi_toolkit/fixtures.clj:
```diff
@@ -0,0 +1,74 @@
+        (.pageLoadTimeout (.. (:webdriver a-driver) manage timeouts) timeout-ms TimeUnit/MILLISECONDS) ;meh meh meh
+        (taxi/implicit-wait 200)                            ;meh
+        (try
+          (taxi/window-resize {:width 1680 :height 1050})
```
Could we make it a parameter?

tlipski replied:
Sure

tlipski replied:
but that makes a lot of options for a simple fixture, so I think we need to work on the structure of fixtures a little bit

tlipski committed:
adjust exit code to be positive to coher with UNIX conventions better

tlipski committed:
extract sub functions from remote-session-fixture

tlipski committed:
updated documentation for following changes:
 - jvm timeout and remote webdriver fixtures
 - fill form enhancements
 - url resolution for environments outside of localhost

kamituel commented:
Thanks :)
